parser/est/est.go

```diff
@@ -27,6 +27,31 @@ type Application struct {
 	Middleware    []*Middleware
 }
 
+// ParamTypes returns all used parameter types used for input to an Encore application
+func (a *Application) ParamTypes() (rtn []*schema.Type) {
+	used := make(map[*schema.Type]struct{})
+
+	// Check the auth handler
+	if a.AuthHandler != nil && a.AuthHandler.Params != nil {
+		used[a.AuthHandler.Params] = struct{}{}
+	}
+
+	// Check all the RPC's
+	for _, s := range a.Services {
+		for _, rpc := range s.RPCs {
+			if rpc.Request != nil {
+				used[rpc.Request.Type] = struct{}{}
+			}
+		}
+	}
+
+	// Convert the set to an array
+	for typ := range used {
```

Comment:
Probably good to return things in a determinstic order. Either sort the `rtn` slice or append to it as we iterate over `Services` and `RPCs` (which are already in deterministic order).